app/src/main/java/br/org/cesar/discordtime/stickysessions/navigation/router/Router.java

```diff
@@ -0,0 +1,38 @@
+package br.org.cesar.discordtime.stickysessions.navigation.router;
+
+import android.app.Activity;
+
+import br.org.cesar.discordtime.stickysessions.R;
+import br.org.cesar.discordtime.stickysessions.navigation.exception.InvalidRouteException;
+import br.org.cesar.discordtime.stickysessions.navigation.repository.IRouteRepository;
+import br.org.cesar.discordtime.stickysessions.navigation.wrapper.IContextWrapper;
+
+public class Router implements IRouter {
+
+    private IContextWrapper mContext;
+    private IRouteRepository mRouteRepository;
+
+    public Router(IRouteRepository routeRepository) {
+        this.mRouteRepository = routeRepository;
+    }
+
+    @Override
+    public void setContext(IContextWrapper context) {
+        mContext = context;
+    }
+
+    @Override
+    public void goNext(String from, String event) throws InvalidRouteException {
+        if(mContext == null){
+            throw new IllegalArgumentException("Null context;" +
+                    "use setContext to set a valid context");
+        }
+        Route route = mRouteRepository.findRoute(from, event);
+        if (route == null) {
+            throw new InvalidRouteException();
```

Comment:
I would add here the "from" and the "event" params on InvalidRouteException message.